# Assignment 2
## Part 1: Running GATK4 Best Practices for Short Variant Discovery
1. Download the demo data for samples S1 and S2 (WGS_data.tar.gz) and the reference script (DNA-analysis-script.sh) from the following link:
   [`Sharepoint Assignment 2`](https://indiana-my.sharepoint.com/:f:/g/personal/ggjiang_iu_edu/Eu3YI73IW4NCo8zBvceaXmkBpAGMv3wRR2ygK3o2oEllxA?e=tu52UQ)
3. Use the demo data to run the GATK4 pipeline.
4. Generate the BAM files for S1 and S2 and VCF file.

## Part 2: Visualization with a Genome Browser
1. Load the BAM files and VCF file for samples S1 and S2 from part 1 into a genome browser.
2. Navigate to chr11 and select a random region that contains a variant and also has sequence(s) in both S1 and S2.
3. Export the selected region as a figure.

## Assignment Submission
An IGV Browser screenshot was generated using alignment files for two samples (S1 and S2) and a VCF file containing identified genetic variants. The VCF file is uploaded above.

The screenshot visualizes alignment and variant calling results for S1 and S2 within the genomic region on chromosome 11, encompassing the C2CD3 gene. Alignment tracks show mismatches and variations, highlighted by colored bars, indicating potential SNPs or indels. The VCF track displays variants, with blue lines for S1 and red lines for S2.

![IGV_screenshot](https://github.com/ssomalra/INFO-B574-Next-Generation-Sequencing/assets/116883221/ad8e5b0d-0d39-4bf1-8b9c-56ffab6b1844)
